File: FORUM/Models/Post.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using System.ComponentModel.DataAnnotations.Schema ;
namespace FORUM.Models
{
    public class Post
    {

         [Key]
         public int Id { get; set; }
         public bool epingler { get; set; }
        
         public int userId  { get; set; }
         public int categoryId { get; set; }

         public string title { get; set; }

         public string content { get; set; }
         public string img { get; set; }

         public string date { get; set; }
         public string views { get; set; }
         public string nbComment { get; set; }

         public int nblike { get; set; }
         public int nbdislike { get; set; }
         [ForeignKey("userId")]
         public User user { get; set; }
         [ForeignKey("categoryId")]
         public Category category { get; set; }
         public List<Comment> comments { get; set; }

        

         




    }
}

File: forum-angular/src/app/DTO/Post.ts
import { Comment } from './Comment';
import { Category } from './Category';
import { User } from './User';
export class Post
{

     public  id :number;
     public  userId :number
     public  title :string
     public  content :string
     public  img :string
     public  date :string
     public  views:string
     public  nbComment:string
     public  categoryId:number

     public  nblike:number
     public  nbdislike:number

     public  user:User

     public  category:Category
     public  comments:Array<Comment>
     public  epingler :boolean



     constructor(){

     }
    //  public List<Comment> comments { get; set; }




}
